zingolib/src/lightclient.rs

```diff
@@ -463,6 +528,123 @@ impl LightClient {
         }
     }
 
+    pub async fn get_user_balances(&self) -> Result<UserBalances, LightWalletError> {
+        let mut balances = UserBalances {
+            spendable: 0,
+            immature_change: 0,
+            minimum_fees: 0,
+            immature_income: 0,
+            dust: 0,
+            incoming: 0,
+            incoming_dust: 0,
+        };
+
+        // anchor height is the highest block height that contains income that are considered spendable.
+        let anchor_height = self.wallet.get_anchor_height().await;
+
+        self
+            .wallet
+            .transactions()
+            .read()
+            .await
+            .current
+            .iter()
+            .for_each(|(_, tx)| {
+                let mature =
+                    !tx.unconfirmed && tx.block_height <= BlockHeight::from_u32(anchor_height);
+                let incoming = tx.is_incoming_transaction();
+
+                let mut change = 0;
+                let mut useful_value = 0;
+                let mut fairy_dust_value = 0;
+                let mut utxo_value = 0;
+                let mut inbound_note_count_nodust = 0;
+                let mut change_note_count = 0;
+
+                tx.orchard_notes
+                    .iter()
+                    .filter(|n| n.spent().is_none() && n.unconfirmed_spent.is_none())
+                    .for_each(|n| {
+                        let value = n.note.value().inner();
+                        if !incoming && n.is_change {
+                            change += value;
+                            change_note_count += 1;
+                        } else if incoming {
+                            if value > MARGINAL_FEE {
+                                useful_value += value;
+                                inbound_note_count_nodust += 1;
+                            } else {
+                                fairy_dust_value += value;
+                            }
+                        }
+                    });
+
+                tx.sapling_notes
+                    .iter()
+                    .filter(|n| n.spent().is_none() && n.unconfirmed_spent.is_none())
+                    .for_each(|n| {
+                        let value = n.note.value().inner();
+                        if !incoming && n.is_change {
+                            change += value;
+                            change_note_count += 1;
+                        } else if incoming {
+                            if value > MARGINAL_FEE {
+                                useful_value += value;
+                                inbound_note_count_nodust += 1;
+                            } else {
+                                fairy_dust_value += value;
+                            }
+                        }
+                    });
+
+                tx.received_utxos
```

Comment:
`TransactionMetadata::received_utxos` is missing in this version of zingolib. What happened to it?

Reply:
Renamed to `transparent_notes`